src/dual.jl

```diff
@@ -384,13 +384,35 @@ for pred in UNARY_PREDICATES
     @eval Base.$(pred)(d::Dual) = $(pred)(value(d))
 end
 
-for pred in BINARY_PREDICATES
+# BINARY_PREDICATES = Symbol[:isequal, :isless, :<, :>, :(==), :(!=), :(<=), :(>=)] # prelude.jl
+
+Base.iszero(x::Dual) = iszero(value(x)) && iszero(partials(x))
```

Comment:
Maybe add a comment that this is just a more efficient implementation and consistent with `x == zero(x)` since `zero(x) = Dual(zero(value(x)), zero(partials(x)))` and `(==)(::Dual, ::Dual)` as defined below?